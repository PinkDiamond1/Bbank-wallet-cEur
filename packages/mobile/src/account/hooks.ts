import { useCallback } from 'react'
import Inquiry, { Environment, InquiryAttributes } from 'react-native-persona'

export function useLaunchPersonaInquiry() {
  const templateId = 'tmpl_zXHYe4JZnuwfMxhzkpmdm45b'
  return useCallback(() => {
    Inquiry.fromTemplate(templateId)
      .environment(Environment.SANDBOX)
      .onSuccess((inquiryId: string, attributes: InquiryAttributes) => {
        console.log(`Inquiry completed for ${inquiryId} with attributes: ${attributes}`)
      })
      .onCancelled(() => {
        console.log('Inquiry #{inquiryId} canceled.')
      })
      .onError((error: Error) => {
        console.error(`Error: ${error.message}`)
      })
      .build()
      .start()
  }, [templateId])
}
