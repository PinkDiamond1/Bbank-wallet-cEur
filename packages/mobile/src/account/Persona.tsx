import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button'
import * as React from 'react'
import { useCallback, useState } from 'react'
import { useAsync } from 'react-async-hook'
import { useTranslation } from 'react-i18next'
import Inquiry, { InquiryAttributes } from 'react-native-persona'
import { useDispatch } from 'react-redux'
import { KycStatus } from 'src/account/reducer'
import { showError } from 'src/alert/actions'
import { ErrorMessages } from 'src/app/ErrorMessages'
import { readOnceFromFirebase } from 'src/firebase/firebase'
import networkConfig from 'src/geth/networkConfig'
import { Namespaces } from 'src/i18n'
import Logger from 'src/utils/Logger'

const TAG = 'PERSONA'

interface Props {
  kycStatus: KycStatus | undefined
}

const Persona = ({ kycStatus }: Props) => {
  const { t } = useTranslation(Namespaces.accountScreen10)
  const [personaAccountCreated, setPersonaAccountCreated] = useState(!!kycStatus)

  const dispatch = useDispatch()

  const templateIdResponse = useAsync(async () => readOnceFromFirebase('persona/templateId'), [])
  const templateId = templateIdResponse.result

  const launchPersonaInquiry = useCallback(() => {
    if (typeof templateId !== 'string') {
      Logger.error(TAG, `Attempted to initiate Persona with invalid templateId: ${templateId}`)
      return
    }

    Inquiry.fromTemplate(templateId)
      .environment(networkConfig.personaEnvironment)
      .onSuccess((inquiryId: string, attributes: InquiryAttributes) => {
        console.log(`Inquiry completed for ${inquiryId} with attributes: ${attributes}`)
      })
      .onCancelled(() => {
        console.log('Inquiry #{inquiryId} canceled.')
      })
      .onError((error: Error) => {
        console.error(`Error: ${error.message}`)
      })
      .build()
      .start()
  }, [templateId])

  useAsync(async () => {
    if (!personaAccountCreated) {
      const response = await fetch(`${networkConfig.inhouseLiquditiyUrl}/persona/account/create`)
      if (response.status === 201 || response.status === 409) {
        setPersonaAccountCreated(true)
      } else {
        dispatch(showError(ErrorMessages.PERSONA_ACCOUNT_ENDPOINT_FAIL))
      }
    }
  }, [])

  return (
    <Button
      onPress={launchPersonaInquiry}
      text={t('raiseLimitBegin')}
      type={BtnTypes.PRIMARY}
      size={BtnSizes.FULL}
      testID="PersonaButton"
      disabled={!personaAccountCreated || !templateId}
    />
  )
}

export default Persona
