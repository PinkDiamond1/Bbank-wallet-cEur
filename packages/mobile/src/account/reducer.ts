import { isE164Number } from '@celo/utils/lib/phoneNumbers'
import { Actions, ActionTypes } from 'src/account/actions'
import { DAYS_TO_DELAY } from 'src/backup/utils'
import { DEV_SETTINGS_ACTIVE_INITIALLY } from 'src/config'
import { features } from 'src/flags'
import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
import Logger from 'src/utils/Logger'
import { ONE_DAY_IN_MILLIS } from 'src/utils/time'
import { Actions as Web3Actions, ActionTypes as Web3ActionTypes } from 'src/web3/actions'

export interface State {
  name: string | null
  e164PhoneNumber: string | null
  pictureUri: string | null
  defaultCountryCode: string | null
  contactDetails: UserContactDetails
  devModeActive: boolean
  devModeClickCount: number
  photosNUXClicked: boolean
  pincodeType: PincodeType
  isSettingPin: boolean
  backupCompleted: boolean
  accountCreationTime: number
  backupRequiredTime: number | null
  dismissedGetVerified: boolean
  dismissedGoldEducation: boolean
  promptFornoIfNeeded: boolean
  retryVerificationWithForno: boolean
  acceptedTerms: boolean
  hasMigratedToNewBip39: boolean
  choseToRestoreAccount: boolean | undefined
  profileUploaded: boolean | undefined
  recoveringFromStoreWipe: boolean | undefined
  accountToRecoverFromStoreWipe: string | undefined
  kycStatus: KycStatus | undefined
}

export enum PincodeType {
  Unset = 'Unset',
  CustomPin = 'CustomPin',
}

export interface UserContactDetails {
  contactId: string | null
  thumbnailPath: string | null
}

export enum KycStatus {
  AccountCreated = 'AccountCreated',
  Verified = 'Verified',
  Denied = 'Denied',
  Pending = 'Pending',
}

export const initialState: State = {
  name: null,
  e164PhoneNumber: null,
  pictureUri: null,
  defaultCountryCode: null,
  contactDetails: {
    contactId: null,
    thumbnailPath: null,
  },
  devModeActive: DEV_SETTINGS_ACTIVE_INITIALLY,
  devModeClickCount: 0,
  photosNUXClicked: false,
  pincodeType: PincodeType.Unset,
  isSettingPin: false,
  accountCreationTime: 99999999999999,
  backupRequiredTime: null,
  backupCompleted: false,
  dismissedGetVerified: false,
  dismissedGoldEducation: false,
  promptFornoIfNeeded: false,
  acceptedTerms: false,
  retryVerificationWithForno: features.VERIFICATION_FORNO_RETRY,
  hasMigratedToNewBip39: false,
  choseToRestoreAccount: false,
  profileUploaded: false,
  recoveringFromStoreWipe: false,
  accountToRecoverFromStoreWipe: undefined,
  kycStatus: undefined,
}

export const reducer = (
  state: State | undefined = initialState,
  action: ActionTypes | RehydrateAction | Web3ActionTypes
): State => {
  switch (action.type) {
    case REHYDRATE: {
      const rehydratedPayload = getRehydratePayload(action, 'account')
      // Ignore some persisted properties
      return {
        ...state,
        ...rehydratedPayload,
        dismissedGetVerified: false,
      }
    }
    case Actions.CHOOSE_CREATE_ACCOUNT:
      return {
        ...state,
        choseToRestoreAccount: false,
      }
    case Actions.CHOOSE_RESTORE_ACCOUNT:
      return {
        ...state,
        choseToRestoreAccount: true,
      }
    case Actions.START_STORE_WIPE_RECOVERY:
      return {
        ...state,
        choseToRestoreAccount: true,
        recoveringFromStoreWipe: true,
        accountToRecoverFromStoreWipe: action.accountToRecover,
        pincodeType: PincodeType.CustomPin,
        acceptedTerms: true,
      }
    case Actions.CANCEL_CREATE_OR_RESTORE_ACCOUNT:
      return {
        ...state,
        choseToRestoreAccount: false,
        recoveringFromStoreWipe: false,
        pincodeType: PincodeType.Unset,
        isSettingPin: false,
      }
    case Actions.SET_NAME:
      return {
        ...state,
        name: action.name,
      }
    case Actions.SET_PICTURE:
      return {
        ...state,
        pictureUri: action.pictureUri,
      }
    case Actions.SAVE_NAME_AND_PICTURE:
      return {
        ...state,
        name: action.name,
        pictureUri: action.pictureUri,
      }
    case Actions.SET_PHONE_NUMBER:
      if (!isE164Number(action.e164PhoneNumber)) {
        return state
      }
      return {
        ...state,
        e164PhoneNumber: action.e164PhoneNumber,
        defaultCountryCode: action.countryCode,
      }
    case Actions.DEV_MODE_TRIGGER_CLICKED:
      const newClickCount = (state.devModeClickCount + 1) % 10
      if (newClickCount === 5) {
        Logger.showMessage('Debug Mode Activated')
      } else if (newClickCount === 0) {
        Logger.showMessage('Debug Mode Deactivated')
      }
      return {
        ...state,
        devModeClickCount: newClickCount,
        devModeActive: newClickCount >= 5,
      }
    case Actions.PHOTOSNUX_CLICKED:
      return {
        ...state,
        photosNUXClicked: true,
      }
    case Actions.SET_PINCODE:
      return {
        ...state,
        isSettingPin: true,
      }
    case Actions.SET_PINCODE_SUCCESS:
      return {
        ...state,
        pincodeType: action.pincodeType,
        isSettingPin: false,
      }
    case Actions.SET_PINCODE_FAILURE:
      return {
        ...state,
        pincodeType: PincodeType.Unset,
        isSettingPin: false,
      }
    case Actions.SET_ACCOUNT_CREATION_TIME:
      return {
        ...state,
        accountCreationTime: action.now,
      }
    case Actions.SET_BACKUP_COMPLETED:
      return {
        ...state,
        backupCompleted: true,
      }
    case Actions.SET_BACKUP_DELAYED:
      return {
        ...state,
        backupRequiredTime: action.now + DAYS_TO_DELAY * ONE_DAY_IN_MILLIS,
      }
    case Actions.TOGGLE_BACKUP_STATE:
      return {
        ...state,
        backupCompleted: !state.backupCompleted,
        backupRequiredTime: null,
      }
    case Actions.DISMISS_GET_VERIFIED:
      return {
        ...state,
        dismissedGetVerified: true,
      }
    case Actions.DISMISS_GOLD_EDUCATION:
      return {
        ...state,
        dismissedGoldEducation: true,
      }
    case Actions.SET_USER_CONTACT_DETAILS:
      return {
        ...state,
        contactDetails: {
          contactId: action.contactId,
          thumbnailPath: action.thumbnailPath,
        },
      }
    case Actions.SET_PROMPT_FORNO:
      return {
        ...state,
        promptFornoIfNeeded: action.promptIfNeeded,
      }
    case Actions.SET_RETRY_VERIFICATION_WITH_FORNO:
      return {
        ...state,
        retryVerificationWithForno: action.retry,
      }
    case Actions.ACCEPT_TERMS: {
      return { ...state, acceptedTerms: true }
    }
    case Actions.UPDATE_KYC_STATUS:
      return {
        ...state,
        kycStatus: action.kycStatus,
      }
    case Web3Actions.SET_ACCOUNT: {
      return {
        ...state,
        hasMigratedToNewBip39: true,
      }
    }
    case Actions.PROFILE_UPLOADED: {
      return {
        ...state,
        profileUploaded: true,
      }
    }
    default:
      return state
  }
}
