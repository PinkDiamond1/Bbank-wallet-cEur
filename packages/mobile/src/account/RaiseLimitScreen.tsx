import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button'
import colors from '@celo/react-components/styles/colors'
import fontStyles from '@celo/react-components/styles/fonts'
import variables from '@celo/react-components/styles/variables'
import React, { useMemo } from 'react'
import { Trans, useTranslation } from 'react-i18next'
import { StyleSheet, Text, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import { useDispatch } from 'react-redux'
import { sendEmail } from 'src/account/emailSender'
import Persona from 'src/account/Persona'
import { KycStatus } from 'src/account/reducer'
import { cUsdDailyLimitSelector, kycStatusSelector } from 'src/account/selectors'
import { showError, showMessage } from 'src/alert/actions'
import { ErrorMessages } from 'src/app/ErrorMessages'
import CurrencyDisplay from 'src/components/CurrencyDisplay'
import { DailyLimitIcon } from 'src/components/FeeIcon'
import { CELO_SUPPORT_EMAIL_ADDRESS } from 'src/config'
import i18n, { Namespaces } from 'src/i18n'
import ApprovedIcon from 'src/icons/ApprovedIcon'
import DeniedIcon from 'src/icons/DeniedIcon'
import InProgressIcon from 'src/icons/InProgressIcon'
import { headerWithBackButton } from 'src/navigator/Headers'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import useSelector from 'src/redux/useSelector'
import { getRecentPayments } from 'src/send/selectors'
import { dailyAmountRemaining } from 'src/send/utils'
import { Currency } from 'src/utils/currencies'
import { accountAddressSelector } from 'src/web3/selectors'
import { Logger } from 'walletconnect-v2/types'

const RaiseLimitScreen = () => {
  const { t } = useTranslation(Namespaces.accountScreen10)
  const dailyLimit = useSelector(cUsdDailyLimitSelector)
  const kycStatus = useSelector(kycStatusSelector)
  const numberIsVerified = true
  // todo: lisa modify ^ to not hard code it as true
  const recentPayments = useSelector(getRecentPayments)
  const accountAddress = useSelector(accountAddressSelector)

  const dispatch = useDispatch()

  const kycAttemptAllowed = !kycStatus || kycStatus === KycStatus.Created

  const sendSupportEmail = async () => {
    try {
      await sendEmail({
        subject: t('kycEmailSubject'),
        recipients: [CELO_SUPPORT_EMAIL_ADDRESS],
        body: t('kycEmailBody', { address: accountAddress }),
        isHTML: true,
      })

      dispatch(showMessage(t('kycEmailSuccess')))
    } catch (error) {
      dispatch(showError(ErrorMessages.KYC_EMAIL_NOT_SENT))
      Logger.error('Error sending indentity verification email', error)
    }
  }

  const applicationStatusTexts = useMemo(() => {
    if (!kycStatus || kycStatus === KycStatus.Created) {
      return null
    }

    return {
      [KycStatus.Pending]: {
        title: t('applicationInReview'),
        description: t('applicationInReviewDescription'),
        icon: <InProgressIcon />,
      },
      [KycStatus.Failed]: {
        title: t('applicationDenied'),
        description: t('applicationDeniedDescription'),
        icon: <DeniedIcon />,
      },
      [KycStatus.Completed]: {
        title: t('applicationCompleted'),
        description: t('applicationCompletedDescription'),
        icon: <ApprovedIcon />,
      },
      [KycStatus.Expired]: {
        title: t('applicationCompleted'),
        description: t('applicationCompletedDescription'),
        icon: <DeniedIcon />, // TODO Lisa: change the title and description here
      },
    }[kycStatus]
  }, [kycStatus])

  const renderButton = () => {
    if (kycStatus === KycStatus.Failed) {
      return (
        <Button
          onPress={sendSupportEmail}
          text={t('contactSupport')}
          type={BtnTypes.SECONDARY}
          size={BtnSizes.FULL}
          style={styles.button}
          testID="ContactSupportButton"
        />
      )
    }
    console.log('lisa kycStatus', kycStatus, numberIsVerified)
    if (!kycStatus || kycStatus === KycStatus.Created) {
      console.log('lisa numberIsVerified')
      return numberIsVerified ? (
        <Persona kycStatus={kycStatus} />
      ) : (
        <Button
          onPress={() => navigate(Screens.VerificationEducationScreen)}
          text={t('raiseLimitConfirmNumber')}
          type={BtnTypes.SECONDARY}
          size={BtnSizes.FULL}
          style={styles.button}
          testID="ConfirmNumberButton"
        />
      )
    }

    return null
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.dailyLimitContainer} testID="DailyLimitContainer">
        <Text style={styles.labelText}>{t('dailyLimitLabel')}</Text>
        {kycStatus === KycStatus.Completed ? (
          <Text style={styles.dailyLimit}>{t('noDailyLimit')} </Text>
        ) : (
          <>
            <View style={styles.dailyLimitDisplay}>
              <CurrencyDisplay
                amount={{ value: dailyLimit, currencyCode: Currency.Dollar }}
                style={styles.dailyLimit}
              />
              <DailyLimitIcon />
            </View>
            <Text style={styles.dailyLimitSubtext}>{t('dailyLimitValue', { dailyLimit })}</Text>
            <Text style={styles.bodyText}>
              <Trans i18nKey={'dailyLimitExplainer'} ns={Namespaces.accountScreen10}>
                <CurrencyDisplay
                  amount={{
                    value: dailyAmountRemaining(Date.now(), recentPayments, dailyLimit),
                    currencyCode: Currency.Dollar,
                  }}
                />
              </Trans>
            </Text>
            {kycAttemptAllowed && (
              <Text style={styles.bodyText}>
                {numberIsVerified ? t('verifyIdentityToRaiseLimit') : t('verifyNumberToRaiseLimit')}
              </Text>
            )}
          </>
        )}
      </View>
      {applicationStatusTexts && (
        <>
          <View style={styles.separator} />
          <View style={styles.applicationStatusContainer}>
            {applicationStatusTexts.icon}
            <Text style={styles.applicationStatusTitle} testID="ApplicationStatus">
              {applicationStatusTexts.title}
            </Text>
          </View>
          <Text style={styles.bodyText}>{applicationStatusTexts.description}</Text>
        </>
      )}
      {renderButton()}
    </SafeAreaView>
  )
}

RaiseLimitScreen.navOptions = () => ({
  ...headerWithBackButton,
  headerTitle: i18n.t('accountScreen10:accountSendLimit'),
})

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginHorizontal: variables.contentPadding,
  },
  dailyLimitContainer: {
    marginBottom: 24,
  },
  labelText: {
    ...fontStyles.label,
    color: colors.gray4,
    marginBottom: 8,
  },
  dailyLimit: {
    ...fontStyles.mediumNumberBold,
  },
  dailyLimitSubtext: {
    ...fontStyles.small500,
    marginTop: 4,
    marginBottom: 24,
  },
  dailyLimitDisplay: {},
  bodyText: {
    ...fontStyles.small,
    marginTop: 8,
  },
  separator: {
    width: '100%',
    height: 1,
    backgroundColor: colors.gray2,
    marginBottom: 16,
  },
  applicationStatusContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 8,
  },
  applicationStatusTitle: {
    ...fontStyles.regular500,
    marginLeft: 6,
  },
  button: {
    marginVertical: 24,
  },
})

export default RaiseLimitScreen
