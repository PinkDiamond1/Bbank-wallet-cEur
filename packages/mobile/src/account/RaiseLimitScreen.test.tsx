import Button from '@celo/react-components/components/Button'
import * as React from 'react'
import 'react-native'
import { fireEvent, render } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import * as renderer from 'react-test-renderer'
import { sendEmail } from 'src/account/emailSender'
import RaiseLimitScreen from 'src/account/RaiseLimitScreen'
import { KycStatus } from 'src/account/reducer'
import { DEFAULT_DAILY_PAYMENT_LIMIT_CUSD } from 'src/config'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { PaymentInfo } from 'src/send/reducers'
import { createMockStore } from 'test/utils'

const createStore = (
  numberVerified: boolean,
  kycStatus?: KycStatus,
  recentPayments: PaymentInfo[] = []
) =>
  createMockStore({
    app: { numberVerified },
    account: { kycStatus },
    send: { recentPayments },
  })

jest.mock('src/firebase/firebase')
jest.mock('src/account/emailSender')

describe('RaiseLimitScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={createStore(true)}>
        <RaiseLimitScreen />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it("shows the 'Confirm Number' button and navigate to the Phone Confirmation flow when the user hasn't completed KYC and doesn't have a confirmed phonen number", async () => {
    const { queryByTestId, getByTestId } = render(
      <Provider store={createStore(false)}>
        <RaiseLimitScreen />
      </Provider>
    )

    expect(queryByTestId('ConfirmNumberButton')).toBeDefined()
    fireEvent.press(getByTestId('ConfirmNumberButton'))
    expect(navigate).toHaveBeenCalledWith(Screens.VerificationEducationScreen)
  })

  it("shows the Persona 'Verify Identity' button when the user hasn't completed KYC and has a confirmed phone number", async () => {
    const { queryByTestId } = render(
      <Provider store={createStore(true, KycStatus.Created)}>
        <RaiseLimitScreen />
      </Provider>
    )

    expect(queryByTestId('PersonaButton')).toBeDefined()
  })

  it("shows the 'Contact Support' button and perpares an email to be sent when the user's KYC submission has been denied", async () => {
    const { queryByTestId, getByTestId, queryByText } = render(
      <Provider store={createStore(true, KycStatus.Failed)}>
        <RaiseLimitScreen />
      </Provider>
    )

    expect(queryByText('Verification Denied')).toBeDefined()
    expect(queryByTestId('ContactSupportButton')).toBeDefined()
    fireEvent.press(getByTestId('ContactSupportButton'))
    expect(sendEmail).toHaveBeenCalled()
  })

  it('shows correct UI when the user has completed KYC', async () => {
    const tree = render(
      <Provider store={createStore(true, KycStatus.Completed)}>
        <RaiseLimitScreen />
      </Provider>
    )

    // Current Daily Limit should be `unlimited`
    expect(tree.queryByText('Unlimited')).toBeDefined()
    expect(tree.queryByType(Button)).toBeNull()
    expect(tree.queryByText('Verification Complete')).toBeDefined()
  })

  it('shows correct UI when the user has pending KYC', async () => {
    const tree = render(
      <Provider store={createStore(true, KycStatus.Pending)}>
        <RaiseLimitScreen />
      </Provider>
    )

    expect(tree.queryByType(Button)).toBeNull()
    expect(tree.queryByText('Verification In Progress')).toBeDefined()
  })

  it('shows the correct remaining transfer limit when a user has made no recent payments', async () => {
    const tree = render(
      <Provider store={createStore(true)}>
        <RaiseLimitScreen />
      </Provider>
    )

    expect(tree.queryByText(DEFAULT_DAILY_PAYMENT_LIMIT_CUSD.toFixed(2))).toBeDefined()
  })

  it('shows the correct remaining transfer limit when a user has made recent payments', async () => {
    const recentPaymentAmount = 100

    const tree = render(
      <Provider
        store={createStore(true, undefined, [
          {
            timestamp: Date.now(),
            amount: recentPaymentAmount,
          },
        ])}
      >
        <RaiseLimitScreen />
      </Provider>
    )

    expect(
      tree.queryByText((DEFAULT_DAILY_PAYMENT_LIMIT_CUSD - recentPaymentAmount).toFixed(2))
    ).toBeDefined()
  })

  it("shows 'no limit' when a user has completed KYC ", async () => {
    const recentPaymentAmount = 100

    const tree = render(
      <Provider
        store={createStore(true, KycStatus.Completed, [
          {
            timestamp: Date.now(),
            amount: recentPaymentAmount,
          },
        ])}
      >
        <RaiseLimitScreen />
      </Provider>
    )

    expect(tree.queryByText('noDailyLimit')).toBeDefined()
  })
})
