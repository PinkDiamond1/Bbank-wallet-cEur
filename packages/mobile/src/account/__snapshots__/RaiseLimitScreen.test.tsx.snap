// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaiseLimitScreen renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 24,
      }
    }
    testID="DailyLimitContainer"
  >
    <Text
      style={
        Object {
          "color": "#9CA4A9",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 13,
          "lineHeight": 16,
          "marginBottom": 8,
        }
      }
    >
      dailyLimitLabel
    </Text>
    <View
      style={Object {}}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 24,
              "lineHeight": 27,
            },
            Object {
              "color": "#2E3338",
            },
          ]
        }
        testID="undefined/value"
      >
        
        $
        1,330.00
      </Text>
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackgroundBorderless",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "paddingHorizontal": 5,
          }
        }
      >
        <svg
          fill="none"
          height={12}
          style={Object {}}
          viewBox="0 0 16 16"
          width={12}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="8"
            r="7"
            stroke="#2E3338"
            stroke-width="1.25"
            style={Object {}}
          />
          <path
            d="M8 12V7M8 6V5"
            stroke="#2E3338"
            stroke-width="1.25"
            style={Object {}}
          />
        </svg>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        statusBarTranslucent={true}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
        <View
          forwardedRef={[Function]}
          hideModalContentWhileAnimating={false}
          onModalHide={[Function]}
          onModalWillHide={[Function]}
          onModalWillShow={[Function]}
          pointerEvents="box-none"
          scrollHorizontal={false}
          scrollOffset={0}
          scrollOffsetMax={0}
          scrollTo={null}
          statusBarTranslucent={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          swipeThreshold={100}
        >
          <RNCSafeAreaView>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "padding": 16,
                  },
                  Object {
                    "borderRadius": 8,
                  },
                  Object {
                    "elevation": 12,
                    "shadowColor": "rgba(156, 164, 169, 0.4)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 12,
                  },
                  Array [
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "maxHeight": "100%",
                      "padding": 24,
                    },
                    undefined,
                  ],
                ]
              }
            >
              <RCTScrollView
                contentContainerStyle={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <View>
                  <Text
                    style={
                      Object {
                        "color": "#2E3338",
                        "fontFamily": "Jost-Medium",
                        "fontSize": 22,
                        "lineHeight": 28,
                        "marginBottom": 12,
                        "textAlign": "center",
                      }
                    }
                  >
                    dailyLimitModal.header
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#2E3338",
                        "fontFamily": "Inter-Regular",
                        "fontSize": 16,
                        "lineHeight": 22,
                        "marginBottom": 24,
                        "textAlign": "center",
                      }
                    }
                  >
                    dailyLimitModal.body
                  </Text>
                </View>
              </RCTScrollView>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-around",
                    "maxWidth": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  nativeBackgroundAndroid={
                    Object {
                      "attribute": "selectableItemBackgroundBorderless",
                      "rippleRadius": undefined,
                      "type": "ThemeAttrAndroid",
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#1AB775",
                          "fontFamily": "Inter-SemiBold",
                          "fontSize": 16,
                          "lineHeight": 22,
                        },
                        Object {
                          "paddingTop": 16,
                        },
                      ]
                    }
                  >
                    global:dismiss
                  </Text>
                </View>
              </View>
            </View>
          </RNCSafeAreaView>
        </View>
      </Modal>
    </View>
    <Text
      style={
        Object {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
          "marginBottom": 24,
          "marginTop": 4,
        }
      }
    >
      dailyLimitValue, {"dailyLimit":1000}
    </Text>
    <Text
      style={
        Object {
          "color": "#2E3338",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "lineHeight": 18,
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            undefined,
            Object {
              "color": undefined,
            },
          ]
        }
        testID="undefined/value"
      >
        
        $
        1,330.00
      </Text>
    </Text>
    <Text
      style={
        Object {
          "color": "#2E3338",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "lineHeight": 18,
          "marginTop": 8,
        }
      }
    >
      verifyIdentityToRaiseLimit
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 100,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#97DFC1",
            "flexGrow": 1,
            "height": 48,
            "justifyContent": "center",
            "opacity": undefined,
            "paddingHorizontal": 24,
            "paddingVertical": 5,
          }
        }
        testID="PersonaButton"
      >
        <Text
          maxFontSizeMultiplier={1}
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          raiseLimitBegin
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
